src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalMessagesOpt.java

```diff
@@ -48,19 +49,19 @@ public void getComponentMessages(MessageCacheItem cacheItem) {
 		try {
 			String filePath = FormatUtils.format(OFFLINE_RESOURCE_PATH, dto.getComponent(), bestMatch.toLanguageTag());
 			Path path = Paths.get(VIPCfg.getInstance().getOfflineResourcesBaseUrl(), filePath);
```

Comment:
Hi Huihui, I think you are talking about a different issue/PR which will be resolved separately.
issue:https://github.com/vmware/singleton/issues/755 
PR: https://github.com/vmware/singleton/pull/858

There is no change in this PR regarding configuring the offlineResourceBaseURL. This bug/PR assumes that offlineResourceBaseURL is still configured in the main VIPCfg config, globally shared across main and "sub" Instances. In other words, the issue above is not related to this. 

The bug is that only the offlineResourceBaseURL folder (again, configured globally in main cfg) inside the product's main file system is loaded. The offlineResourceBaseURL folder inside the subinstance (shared lib) jar is not read. The expected behavior is that it will also load the bundles inside the offlineResourceBaseURL folder in the shared library jars. In other words, if the the shared library jar carries the translations inside it, the translation should be loaded by the root application.

Reply:
On a side note that is not related to the scope of this PR, we will resolve your concern about being able to configure separate VIPCfg properties per product/shared lib in issue:https://github.com/vmware/singleton/issues/755. You are headed towards the right direction though. VIPCfg.getInstance().. is not correctly used. Once we remove this singleton pattern implementation through PR: https://github.com/vmware/singleton/pull/858, instead of VIPCfg.getInstance(), we will start using VIPCfgFactory.getCfg(productname) for product-specific cases such as L3, and VIPCfgFactory.getCfg() for non-product-specific cases such as for patterns. See the samples in the PR and my reply to your comments there for your reference. Again, this PR is not related to that.